### PR TITLE
Add label selectors to informer's cache

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- the controller manager is created with a cache, which uses label selector for Secrets and Config Maps to reduce the risk of OOM

**Related issue(s)**

### CODE REVIEW

File: controllers/cache.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+	labelSelector, err := labels.Parse("app.kubernetes.io/managed-by in (btp-manager,kcp-kyma-environment-broker)")
+	if err != nil {
+		panic(fmt.Sprintf("unable to parse label selector: %s", err))
+	}
+	objSelector := cache.ObjectSelector{
+		Label: labelSelector,
+	}
+
+	opts.SelectorsByObject = map[client.Object]cache.ObjectSelector{
+		&corev1.Secret{}:    objSelector,
```
Comment (PK85) [inline controllers/cache.go:22]: Have you limited that to secrets and configmaps only by purpose? I mean we have webhooks too.

**Review by PK85 — CHANGES_REQUESTED** [review]
Add a new document about controller cache. Mention how we build cache and why

File: scripts/testing/create-secrets-configmaps.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply scripts/testing/create-secrets-configmaps.sh:1]: will be done in separate PR

**Review by PK85 — APPROVED** [review]

File: docs/contributor/07-10-informer-cache.md [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# Informer's cache
+
+The controller manager uses informers with a cache. All observed resources (btpoperator, secret, configmap etc.) are stored in the cache. Because of out of memory risk, the cache is configured with a label selector `app.kubernetes.io/managed-by in (btp-manager,kcp-kyma-environment-broker)`. See [cache.go](/controllers/cache.go).
```
Comment (IwonaLanger) [inline docs/contributor/07-10-informer-cache.md:3]: The phrase "the out of memory risk" should be changed to:
- the out-of-memory risk if 'out-of-memory' just describes the type of risk
- the "out of memory" risk if 'out of memory' is the name used for the risk

File: scripts/testing/create-secrets-configmaps.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# Stress testing in regard to memory consumption - could cause OOM (but should not).
+# Creates btp-operator and numerous service instances and service bindings in current context.
```
Comment (IwonaLanger) [inline scripts/testing/create-secrets-configmaps.sh:3]:
```suggestion
# Creates btp-operator and numerous Service Instances and Service Bindings in the current context.
```

File: scripts/testing/create-secrets-configmaps.sh [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+#
+# The script has the following arguments:
+#     - number of config maps and secrets (the provided number is multiplied by 10)
```
Comment (IwonaLanger) [inline scripts/testing/create-secrets-configmaps.sh:6]:
```suggestion
#     - number of Config Maps and Secrets (the provided number is multiplied by 10)
```

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**